pkg/kubelet/kubelet_pods.go

```diff
@@ -343,13 +343,19 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 		klog.V(5).InfoS("Mount has propagation", "pod", klog.KObj(pod), "containerName", container.Name, "volumeMountName", mount.Name, "propagation", propagation)
 		mustMountRO := vol.Mounter.GetAttributes().ReadOnly
 
+		rro, err := resolveRecursiveReadOnly(mount, supportsRRO, utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts))
```

Comment:
This is not the normal pattern for gates - is there a reason?  Usually we would check the gate and wipe the field in  pkg/api/pod/util.go `dropDisabledFields()`.

Reply:
Also, the gate needs to control ALL of the new fields, including status.

Reply:
`resolveRecursiveReadOnly` takes a gate as an argument for ease of unit testing

Reply:
First and foremost, the gate prevents the API from being used by net-new objects.  When a user specifies RRO and the gate is disabled, we erase the field and pretend we don't know about it.  If RRO was already in use, we allow it to stay in use.

It's a secondary question whether we actually respect RRO.  Superficially, it seems reasonable to say "I know it says RRO, but that was from before, and we're not going to respect that".

Reply:
Updated `dropDisabledFields`